ibis/spark/compiler.py

```diff
@@ -0,0 +1,49 @@
+import ibis.sql.compiler as comp
+from ibis.bigquery.compiler import (
+    BigQueryContext,
+    BigQueryExprTranslator,
+    BigQueryTableSetFormatter,
+    BigQueryUnion,
+)
+from ibis.impala import compiler as impala_compiler
+
+
+class SparkSelectBuilder(comp.SelectBuilder):
+    @property
+    def _select_class(self):
+        return SparkSelect
+
+
+SparkUnion = BigQueryUnion
+
+
+class SparkQueryBuilder(comp.QueryBuilder):
+    select_builder = SparkSelectBuilder
+    union_class = SparkUnion
+
+
+def build_ast(expr, context):
+    builder = SparkQueryBuilder(expr, context=context)
+    return builder.get_result()
+
+
+SparkContext = BigQueryContext
```

Comment:
Don't mix bigquery things and impala things. You should just be able to use the impala versions of these for now.